### PR TITLE
OpenAPI CRD user guide doc

### PR DESCRIPTION
**Pull request by eguzki**
Explained how 3scale is configured from the OpenAPI document. Which parts of the OpenAPI document are used, and how, to configure 3scale.

Also added documentation about the minimum OpenAPI Document requried to configure a working 3scale product.

https://issues.redhat.com/browse/THREESCALE-4712

### CODE REVIEW

File: doc/openapi-user-guide.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      * [3scale Deployment Mode](#3scale-deployment-mode)
+   * [Minimum required OAS doc](#minimum-required-oas-doc)
+   * [Link your OpenAPI spec to your 3scale tenant or provider account](#link-your-openapi-spec-to-your-3scale-tenant-or-provider-account)
+
+Generated using [github-markdown-toc](https://github.com/ekalinin/github-markdown-toc)
+
+## OpenAPI document sources
+
+The OpenAPI document <OAS> can be read from different sources:
+* Openshit secret
```
Comment (miguelsorianod) [inline doc/openapi-user-guide.md:32]: Typo(:eyes:) and it should be `Kubernetes` secret

File: doc/openapi-user-guide.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+3scale policy chain will be the default one created by 3scale.
+
+### 3scale Deployment Mode
+
+By default, the configured 3scale deployment mode will be `APIcast 3scale managed`.
+However, when the `spec.productionPublicBaseURL` or the `spec.stagingPublicBaseURL` (or both)
+fields are provided in the [OpenAPI custom resource](openapi-reference.md),
+the product's deployment mode will be `APIcast self-managed`.
+
+Example of a [OpenAPI custom resource](openapi-reference.md) with custom public base URL
```
Comment (miguelsorianod) [inline doc/openapi-user-guide.md:184]: Missing `:` at the end

**Review by miguelsorianod — CHANGES_REQUESTED** [review]
:+1: , just a couple minor comments

**Comment by eguzki** [comment]
requested changes applied.

**Review by miguelsorianod — APPROVED** [review]

**Pull request merged**